### PR TITLE
GeoJSON CSV support

### LINKED ISSUE #134: Attributes Text data with spaces not stored properly
If you create a text attribute as a list of different options:
image
Then set an attribute to have one of them:
image
It saves properly in the local viewer and it looks like it is sent properly as well to the server:
image
Data retrieved back from the server has the wrong key:
image
Consequently it is stored incorrectly for the viewer:
image

### PR DESCRIPTION
**Pull request by BryonLewis**
This is a prerequisite for the layer-splitting head/tails update in the other PR.

Part of this when pushed to the server will require a data migration to change all items that have a `feature.head and feature.tail` to the new geoJSON format.

This should support a single `(poly)` as well as the current `(kp) head` and `(kp) tail` format for the points in the CSV file.

Questions:
I didn't know if we should support true floats for geoJSON coordinates.  They are placed and submitting using floating point numbers (as is everything in JS) so I kept support for these in the event we do have sub pixel placement as a result of utilizing the zooming feature.

Fixed:
- Start point for the processing of the row in a track was off and would only allow processing of at most two additional `(kp) or (poly) or (atr)` options.  We know that the base size of the CSV is 9 columns with a + (2* # of confidence pairs), so I use that as a starting point instead.
- Track attributes had an issue with supporting more than one.  I've updated that by adding `.items()` to the for loop.  Seemed to fix it and allow more than one.
- Fixes #134 -  There was an issue with Track attributes having issues with string types that had spaces in them.  This seemed to be a result of the regex used to separate out the track names.  It now clearly looks for a space and takes all data after the space at the attribute. 
```
original:            groups = re.match(r"\(atr\) (.+) (.+)", row[j])
updated:            groups = re.match(r"\(atr\) (.*?)\s(.+)", row[j])
```

### CODE REVIEW

**Comment by subdavis** [comment]
I'm working on some unit tests for master that we should include in these.  Since the migration script isn't ready yet, I don't think I'm blocking you by doing that.

**Comment by subdavis** [comment]
https://github.com/VIAME/VIAME-Web/pull/308 provides the framework for tests.   Once that merges, we should rebase and add additional tests for the new parts of the schema.

If this is ready before that, we can just create a follow-up.

File: server/viame_server/serializers/viame.py [hunk 1]
```diff
@@ -43,6 +43,35 @@ def _deduceType(value: str) -> Union[bool, float, str]:
...
+        feature = {
+            "type": "Feature",
+            "properties": {
+                "key": key
+            },
+            "geometry":{
+                "type": type
+            },
+        }
+    if "Polygon" in type:
```
[the remark this answers is not in the document]
Comment (BryonLewis) [reply server/viame_server/serializers/viame.py:69]: Yeah I see the problem in the fact that I don't support MultiPolygon and I think we want to keep it that way.  All polygons need to be singular and tied to a key?

File: server/viame_server/serializers/viame.py [hunk 2]
```diff
@@ -221,15 +212,28 @@ def export_tracks_as_csv(
...
-                            if 'Point' in geoJSONFeature.geometry.type:
-                                coordinates  = geoJSONFeature.geometry.coordinates
-                                columns.append(f"(kp) {geoJSONFeature.properties['key']} {coordinates[0]} {coordinates[1]}")        
+                            if 'Polygon' == geoJSONFeature.geometry.type:
+                                # Coordinates need to be flattened out from their list of tuples
+                                coordinates = [
+                                    item
+                                    for sublist in geoJSONFeature.geometry.coordinates[
+                                        0
+                                    ]
```
Comment (subdavis) [inline server/viame_server/serializers/viame.py:226]: what a mess.  Thanks, black....

File: server/viame_server/serializers/viame.py [hunk 3]
```diff
@@ -43,36 +42,34 @@ def _deduceType(value: str) -> Union[bool, float, str]:
...
-                "type": type
-            },
+            "properties": {"key": key},
+            "geometry": {"type": type},
         }
-    if "Polygon" in type:
+    if "Polygon" == type:
         feature["geometry"]['coordinates'] = [coords]
-    else:
+    elif ["LineString", "Point"].count(type):
```
Comment (subdavis) [inline server/viame_server/serializers/viame.py:67]:
```suggestion
    elif type in ["LineString", "Point"]:
```

Pythonic™

**Pull request merged**